README.md

```diff
@@ -27,18 +27,39 @@ Put jsonnet method definitions together in a directory, each named named
 server receives a call to that method.
 
 The request protobuf message is marshaled to JSON and set as the `input` jsonnet
-top-level argument when evaluating the jsonnet method. It is marshaled as:
+top-level argument when evaluating the jsonnet method. For a
+non-client-streaming method, it is marshaled as:
 
     {
         request: { ...json-encoded protobuf here... }
     }
 
-The jsonnet method definition should be a function of the following form:
+For a client-streaming method, it is marshaled as:
+
+    {
+        stream: [ {message1}, {message2}, ...]
+    }
+
+For bidirectional streaming methods, the jsonnet method definition is evaluated
+once for each message on the client stream (input) with the input defined as per
+the non-client-streaming method case above (single message, not array). If there
```

Comment:
This sentence is too long to follow for me.

In contrast to client streaming, for bidirectional streaming methods, the jsonnet method definition is evaluated
once for each message on the client stream. On each evaluation the input object contains a single `request` field and not the `stream` array.

Reply:
rewritten.